packages/aws-cdk-lib/aws-kinesisfirehose/lib/iceberg-destination.ts

```diff
@@ -0,0 +1,366 @@
+
+  /**
+   * The retry duration after a delivery failure.
+   *
+   * @default Duration.seconds(3600)
+   */
+  readonly retryDuration?: Duration;
+
+  /**
+   * Enable automatic schema evolution.
+   *
+   * When enabled, Firehose can automatically evolve the schema of the Iceberg table
+   * to accommodate new fields in the incoming data.
+   *
+   * Note: This feature is only available for DatabaseAsSourceStream and is in preview.
+   * It may not work with standard delivery streams.
+   *
+   * @default false
+   */
+  readonly schemaEvolutionEnabled?: boolean;
+
+  /**
+   * Enable automatic table creation.
+   *
+   * When enabled, Firehose can automatically create Iceberg tables if they don't exist.
+   * Requires warehouseLocation to be specified in catalogConfiguration.
+   *
+   * Note: This feature is only available for DatabaseAsSourceStream and is in preview.
+   * It may not work with standard delivery streams.
+   *
+   * @default false
+   */
+  readonly tableCreationEnabled?: boolean;
+}
+
+/**
+ * An Apache Iceberg destination for data from a Kinesis Data Firehose delivery stream.
+ */
+export class IcebergDestination implements IDestination {
+  constructor(
+    private readonly bucket: s3.IBucket,
+    private readonly props: IcebergDestinationProps,
+  ) {
+    // Validate that at least one catalog configuration is provided
+    if (!props.catalogConfiguration.catalogArn && !props.catalogConfiguration.warehouseLocation) {
+      throw new UnscopedValidationError('IcebergDestination requires at least one of catalogConfiguration.catalogArn or catalogConfiguration.warehouseLocation to be specified');
+    }
+
+    // Validate S3 backup mode - Iceberg only supports FAILED mode
+    if (this.props.s3Backup?.mode === BackupMode.ALL) {
+      throw new UnscopedValidationError('Iceberg destinations only support BackupMode.FAILED for S3 backup');
+    }
+
+    // Validate that warehouseLocation is provided when schema evolution or table creation is enabled
+    if ((props.schemaEvolutionEnabled || props.tableCreationEnabled) && !props.catalogConfiguration.warehouseLocation) {
+      throw new UnscopedValidationError('catalogConfiguration.warehouseLocation is required when schemaEvolutionEnabled or tableCreationEnabled is true');
+    }
+  }
+
+  bind(scope: Construct, _options: DestinationBindOptions): DestinationConfig {
+    // Create or use provided IAM role
+    const role = this.props.role ?? new iam.Role(scope, 'Iceberg Destination Role', {
+      assumedBy: new iam.ServicePrincipal('firehose.amazonaws.com'),
+    });
+
+    // Grant S3 permissions
+    const bucketGrant = this.bucket.grantReadWrite(role);
+
+    // Grant Glue Catalog permissions
+    // Permissions are needed for both catalogArn and warehouseLocation configurations
+    const catalogArn = this.props.catalogConfiguration.catalogArn;
+    const actions = [
+      'glue:GetDatabase',
+      'glue:GetTable',
+      'glue:UpdateTable',
+    ];
```

Comment:
Please add a link to the doc describing these permissions in the code comment.

namely https://docs.aws.amazon.com/firehose/latest/dev/controlling-access.html#using-s3-tables